pkg/volume/testing/testing.go

```diff
@@ -263,7 +263,7 @@ func (fv *FakeVolume) TearDownAt(dir string) error {
 	return os.RemoveAll(dir)
 }
 
-func (fv *FakeVolume) Attach(spec *Spec, hostName string) error {
+func (fv *FakeVolume) Attach(coudProvider cloudprovider.Interface, spec *Spec, hostName string) error {
```

Comment:
I made this comment on the child PR. I guess it is more appropriate here: 

Not all volume plugins will have a cloud provider. So we should not require one in the Attach/Detach interfaces. Plugins that require one should get it through the VolumeHost.